src/vtk_calculator.py

```diff
@@ -1,53 +1,106 @@
-            Syntax is the same as used in the calculator object, coordinates are given as e.g. 'coordX'.""")
-    parser.add_argument("--out", "-o", dest="out_meshname", help="""The output meshname. 
+    parser.add_argument("function",
+                        help="""The function to evalutate on the mesh.
+            Syntax is the same as used in the calculator object, coordinates are given as e.g. 'coordsX+coordsY'.""")
+    parser.add_argument("--out", "-o", dest="out_meshname", default=None, help="""The output meshname.
             Default is the same as for the input mesh""")
-    parser.add_argument("--log", "-l", dest="logging", default="INFO", 
-            choices=["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"], help="""Set the log level. 
+    parser.add_argument("--tag", "-t", dest="tag", default="MyScalar", help="""The tag for output data.
+            Default is MyScalar""")
+    parser.add_argument("--intag", "-it", dest="intag", help="""The tag for input data.
+            Required for diff mod""")      
+    parser.add_argument("--log", "-l", dest="logging", default="INFO",
+                        choices=["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"], help="""Set the log level.
             Default is INFO""")
-
+    parser.add_argument("--diff", "-d", action='store_true', help="Calculate the difference to present data.")
+    parser.add_argument("--stats", "-s", action='store_true',
+                        help="Store stats of the difference calculation as the separate file inputmesh.stats.json")
     args = parser.parse_args()
     args.out_meshname = args.out_meshname or args.in_meshname
     return args
 
+
 def main():
     args = parse_args()
-    logging.basicConfig(level = getattr(logging, args.logging))
+    logging.basicConfig(level=getattr(logging, args.logging))
 
     assert os.path.isfile(args.in_meshname), "Input mesh file not found!"
-    data = mesh_io.read_dataset(args.in_meshname)
-    logging.info("Read in {} points.".format(data.GetNumberOfPoints()))
+
+    out_meshname = args.out_meshname
+    if args.out_meshname is None:
+        out_meshname = args.in_meshname
+
+    reader = vtk.vtkGenericDataObjectReader()
+    reader.SetFileName(args.in_meshname)
+    reader.Update()
+    vtk_dataset = reader.GetOutput()
+    logging.info("Read in {} points.".format(vtk_dataset.GetNumberOfPoints()))
 
     calc = vtk.vtkArrayCalculator()
-    calc.SetInputData(data)
+    calc.SetInputData(vtk_dataset)
     calc.AddCoordinateScalarVariable("coordsX", 0)
     calc.AddCoordinateScalarVariable("coordsY", 1)
     calc.AddCoordinateScalarVariable("coordsZ", 2)
     calc.SetFunction(args.function)
-    calc.SetResultArrayName("scalars")
-    calc.Update()
+    calc.SetResultArrayName(args.tag)
+    calc.Update()    
     logging.info("Evaluated '{}' on the mesh.".format(args.function))
+    
+    if args.diff:
+        assert args.intag is not None, "Difference mode needs input array tag"
```

Comment:
This has to work, as it is the default use-case for the script.

Reply:
It is solved as if it is not given `tag` will be used as `intag`.

```    if args.diff and args.intag is None:
        logging.info("No intag is given outtag '{}' will be used as intag.".format(args.tag))
        intag = args.tag
```